File: packages/core/frontend/src/types/resources/event.ts
import type { ManagedField } from "./shared";

export interface Event {
  Type: string;
  Message: string;
  Namespace: string;
  InvolvedObjectKind: string;
  InvolvedObjectName: string;
  Source: string;
  Count: number;
  Age: string;
  LastSeen: string;
  CreatedAt: number;
  Name: string;
  Reason: string;
  FirstSeen: string;
  FirstSeenAt: number;
  LastSeenAt: number;
  InvolvedObjectFieldPath: string;
  InvolvedObjectNamespace: string;
  ManagedFields: ManagedField[];
}

File: packages/core/kube/dto/event.go
package dto

type Event struct {
	Type                    string
	Message                 string
	Namespace               string
	InvolvedObjectKind      string
	InvolvedObjectName      string
	Source                  string
	Count                   int32
	Age                     string
	LastSeen                string
	CreatedAt               int64
	Name                    string
	Reason                  string
	FirstSeen               string
	FirstSeenAt             int64
	LastSeenAt              int64
	InvolvedObjectFieldPath string
	InvolvedObjectNamespace string
	ManagedFields           []ManagedField
}
